### ISSUE TITLE
Allow `register` with constructor args to specify a configuration action

### ISSUE THREAD

**Issue by lacasseio** [gradle/gradle-native#835]
Following discussion here: https://github.com/gradle/gradle/issues/6524#issuecomment-416202531

We should consider adding an overload like
```
TaskProvider<T> register(String name, Class<T> type, Iteratable<Object> constructorArgs, Action<T> configuration)
```

So you can call the method like:
```
tasks.register("some-task", CustomTask, ["arg1", 42, 52.0]) { /* ... */ }
```

Kolin DSL could have the following overload:
```
fun <T: Task> TaskContainer.register(name: String, type: KClass<T>, vararg constructorArgs: String, configuration: T.() -> Unit): TaskProvider<T> =
    register(name, type.java, constructorArgs).apply { configure(configuration) }
```

That can be used like this:
```
tasks.register("someTask", CustomTask::class, 42, 52.0) { /* … */ }
```

**Comment by big-guy** [comment]
I don't think this works well in practice or at least, any better than two separate lines.  For Groovy/Java, you have to use the explicit list.  For Kotlin, you need to use named arguments because the vararg is not last, I think.

I'm not sure how common the constructor args way of creating tasks is going to be in practice. 

I think all we're replacing is this:
```
val someTask = tasks.register("someTask", CustomTask::class, 42, 52.0) 
someTask.configure { /* … */ }
```

**Comment by bamboo** [comment]
@big-guy, the 2nd line in your snippet can be written without `.configure`:

```kotlin
someTask { /* ... */ }
```

**Comment by big-guy** [comment]
Yep, I thought we could do that, but I didn't check.  Even more of a reason to not add this method :)

**Comment by StefMa** [comment]
The original issues cames exactly because this (what you suggest @big-guy) should/can be avoided.
When we go a step back and don't "want" to use the new `register` overload we could think about the original proposal again:
Return `TaskContainer` in `configure`.
Then we can write this:
```
val someTask = tasks.register("someTask", CustomTask::class, 42, 52.0).configure {
}
```
And that is from my point of view still the best option... 🤔
